File: Unique-Finance-Tracker/scratch/inspect_db.py
import sqlite3

db_path = r'C:\Users\user\Downloads\Unique-Finance-Tracker\Unique-Finance-Tracker\sqlite.db'

try:
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()

    print("--- Tables ---")
    cursor.execute("SELECT name, sql FROM sqlite_master WHERE type='table'")
    for name, sql in cursor.fetchall():
        print(f"Table: {name}")
        print(f"SQL: {sql}")
        print("-------------------")

    print("\n--- Indexes ---")
    cursor.execute("SELECT name, tbl_name, sql FROM sqlite_master WHERE type='index'")
    for name, tbl_name, sql in cursor.fetchall():
        print(f"Index: {name} on {tbl_name}")
        print(f"SQL: {sql}")
        print("-------------------")

    print("\n--- Users Columns ---")
    cursor.execute("PRAGMA table_info(users)")
    for col in cursor.fetchall():
        print(f"Column ID: {col[0]}, Name: {col[1]}, Type: {col[2]}, NotNull: {col[3]}, DefaultValue: {col[4]}")

except Exception as err:
    print("Error inspecting database:", err)
finally:
    if 'conn' in locals():
        conn.close()

File: Unique-Finance-Tracker/scratch/inspect_db.mjs
import Database from 'better-sqlite3';
import path from 'path';

const dbPath = 'C:\\Users\\user\\Downloads\\Unique-Finance-Tracker\\Unique-Finance-Tracker\\sqlite.db';
const db = new Database(dbPath);

try {
    console.log("--- Tables ---");
    const tables = db.prepare("SELECT name, sql FROM sqlite_master WHERE type='table'").all();
    tables.forEach(t => {
        console.log(`Table: ${t.name}`);
        console.log(`SQL: ${t.sql}`);
        console.log("-------------------");
    });

    console.log("\n--- Indexes ---");
    const indexes = db.prepare("SELECT name, tbl_name, sql FROM sqlite_master WHERE type='index'").all();
    indexes.forEach(i => {
        console.log(`Index: ${i.name} on ${i.tbl_name}`);
        console.log(`SQL: ${i.sql}`);
        console.log("-------------------");
    });

    console.log("\n--- Users Columns ---");
    const columns = db.prepare("PRAGMA table_info(users)").all();
    columns.forEach(c => {
        console.log(`Column ID: ${c.cid}, Name: ${c.name}, Type: ${c.type}, NotNull: ${c.notnull}, DefaultValue: ${c.dflt_value}`);
    });

} catch (err) {
    console.error("Error inspecting database:", err);
} finally {
    db.close();
}
